### PR TITLE
UPSTREAM: <carry>: Upgrade go version to 1.21 and go.mod package versions

### PR DESCRIPTION
**Pull request by DharmitD**
**Description of your changes:**
Upgrade go version to 1.21
Resolve go.opentelemetry.io/contrib package related CVEs by upgrading package version to latest
Update License CSVs
Remove follow tags to be in sync with cpaas midstream.

Implemented fixes on my fork, tested them out by importing the fork in my personal snyk account and making sure the CVEs were eliminated.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 0 commits between 2aacfe2781c01939e087e188cdb04a0e117bfb2c...fe059b09fba2abc679beaf8bb375bc6563381e19
```

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 0 commits between 2aacfe2781c01939e087e188cdb04a0e117bfb2c...750bb84227c3f507ec42f40fea6206ea88f0b7d9
```

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 0 commits between 2aacfe2781c01939e087e188cdb04a0e117bfb2c...f860573b99119c7af9001520c2e165a69bd521b4
```

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-56`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-56`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-56`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-56`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-56`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-56`

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 0 commits between 2aacfe2781c01939e087e188cdb04a0e117bfb2c...54482e6db9aaddca67168d87bdf11d2f73ea7c44
```

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-56`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-56`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-56`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-56`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-56`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-56`

File: backend/Dockerfile.driver [hunk 1]
```diff
@@ -16,8 +16,8 @@
 ARG SOURCE_CODE=.
 
 # Use ubi8/nodejs-14 as base image
-#@follow_tag(registry.access.redhat.com/ubi8/go-toolset:1.20)
-FROM registry.access.redhat.com/ubi8/go-toolset:1.20 as builder
+#@follow_tag(registry.access.redhat.com/ubi8/go-toolset:1)
```
Comment (amadhusu) [inline backend/Dockerfile.driver:19]: Doesn't matter much but there is a typo here in the #@follow_tag, it should be `registry.access.redhat.com/ubi8/go-toolset:21` but apart from that tested overall and it looks good.

**Review by amadhusu — APPROVED** [review]

File: backend/Dockerfile.driver (hunk 1 above)
Comment (DharmitD) [reply backend/Dockerfile.driver]: Done, updated it to be `1.21`. Thanks!

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 0 commits between 2aacfe2781c01939e087e188cdb04a0e117bfb2c...2c264cc72d62f6941bccfbf2f01f23b6e35b64a5
```

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-56`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-56`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-56`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-56`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-56`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-56`

File: backend/Dockerfile.viewercontroller [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.20.4-alpine3.17 as builder
+FROM golang:1.21.4-alpine3.17 as builder
```
Comment (HumairAK) [inline backend/Dockerfile.viewercontroller:15]: some dockerfiles look like they weren't updated like launcher, can you go over those again?

Comment (DharmitD) [reply backend/Dockerfile.viewercontroller:15]: done, updated all dockerfiles.

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 2 commits between 2aacfe2781c01939e087e188cdb04a0e117bfb2c...480e9a0c6e210ed64c80c7beb2a7e5366945c4e2

UPSTREAM commit 480e9a0 has invalid summary Update go version references to 1.21.

UPSTREAM commits are validated against the following regular expression:
  ^UPSTREAM: (revert: )?(([\w.-]+/[\w-.-]+)?: )?(\d+:|<carry>:|<drop>:)

UPSTREAM commit summaries should look like:

  UPSTREAM: <PR number|carry|drop>: description

UPSTREAM commits which revert previous UPSTREAM commits should look like:

  UPSTREAM: revert: <normal upstream format>

Examples of valid summaries:

  UPSTREAM: 12345: A kube fix
  UPSTREAM: <carry>: A carried kube change
  UPSTREAM: <drop>: A dropped kube change
  UPSTREAM: revert: 12345: A kube revert


```

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-56`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-56`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-56`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-56`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-56`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-56`

File: backend/Dockerfile.scheduledworkflow [hunk 3]
```diff
@@ -16,8 +16,8 @@
 ARG SOURCE_CODE=.
 
 # Use ubi8/nodejs-14 as base image
-#@follow_tag(registry.access.redhat.com/ubi8/go-toolset:1.20)
-FROM registry.access.redhat.com/ubi8/go-toolset:1.20 as builder
+#@follow_tag(registry.access.redhat.com/ubi8/go-toolset:1.21)
```
Comment (gmfrasca) [inline backend/Dockerfile.scheduledworkflow:19]: `@follow_tags` have been removed from midstream, so we should do it here too

File: backend/Dockerfile.launcher [hunk 4]
```diff
@@ -18,8 +18,8 @@ ARG CI_CONTAINER_VERSION="unknown"
 
 
 # Use ubi8/nodejs-14 as base image
-#@follow_tag(registry.access.redhat.com/ubi8/go-toolset:1.20)
-FROM registry.access.redhat.com/ubi8/go-toolset:1.20 as builder
+#@follow_tag(registry.access.redhat.com/ubi8/go-toolset:1.21)
```
Comment (gmfrasca) [inline backend/Dockerfile.launcher:21]: @follow_tags have been removed from midstream, so we should do it here too

File: backend/Dockerfile.driver [hunk 5]
```diff
@@ -16,8 +16,8 @@
 ARG SOURCE_CODE=.
 
 # Use ubi8/nodejs-14 as base image
-#@follow_tag(registry.access.redhat.com/ubi8/go-toolset:1.20)
-FROM registry.access.redhat.com/ubi8/go-toolset:1.20 as builder
+#@follow_tag(registry.access.redhat.com/ubi8/go-toolset:1.21)
```
Comment (gmfrasca) [inline backend/Dockerfile.driver:19]: @follow_tags have been removed from midstream, so we should do it here too

File: backend/Dockerfile [hunk 6]
```diff
@@ -15,8 +15,8 @@
 # Build arguments
 ARG SOURCE_CODE=.
 
-#@follow_tag(registry.access.redhat.com/ubi8/go-toolset:1.20)
-FROM registry.access.redhat.com/ubi8/go-toolset:1.20 as builder
+#@follow_tag(registry.access.redhat.com/ubi8/go-toolset:1.21)
```
Comment (gmfrasca) [inline backend/Dockerfile:18]: @follow_tags have been removed from midstream, so we should do it here too

**Review by gmfrasca — COMMENTED** [review]
small comment-removal comments, but otherwise /lgtm

**Comment by gmfrasca** [comment]
that said, since you updated `go.mod`, you'll need to update the license CSVs. you can do that by simply running `make all` in `backend/`

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 3 commits between 2aacfe2781c01939e087e188cdb04a0e117bfb2c...f90fb046f7a9b8120334a11ab063220cc2dede83
```

File: backend/Dockerfile.scheduledworkflow (hunk 3 above)
Comment (DharmitD) [reply backend/Dockerfile.scheduledworkflow]: Done, removed tags.

File: backend/Dockerfile.launcher (hunk 4 above)
Comment (DharmitD) [reply backend/Dockerfile.launcher]: Done, removed tags.

File: backend/Dockerfile.driver (hunk 5 above)
Comment (DharmitD) [reply backend/Dockerfile.driver]: Done, removed tags.

File: backend/Dockerfile (hunk 6 above)
Comment (DharmitD) [reply backend/Dockerfile]: Done, removed tags.

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 4 commits between 2aacfe2781c01939e087e188cdb04a0e117bfb2c...40420106906722c7b20708c5dfd7dad8fca9d53e
```

**Comment by DharmitD** [comment]
> that said, since you updated `go.mod`, you'll need to update the license CSVs. you can do that by simply running `make all` in `backend/`

Done, updated the license CSVs.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-56`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-56`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-56`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-56`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-56`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-56`

**Review by gmfrasca — COMMENTED** [review]
/lgtm

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-56`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-56`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-56`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-56`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-56`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-56`

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 1 commits between ea6815018f49bdb884ff5b955b946f64ffc8f648...f46e5106dd958d92ca85f5c2942306a7b1bb4a69
```

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-56`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-56`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-56`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-56`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-56`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-56`

**Comment by dsp-developers** [comment]
### Commit Checker results:

```
**NOTE**: These are the results of the commit checker scans. 
If these are not commits from upstream kfp, then please ensure
you adhere to the commit checker formatting
```
```
commitchecker verson unknown
Validating 0 commits between dadfb38393ce96cb7dc1eac60428d66f0e3f1933...f46e5106dd958d92ca85f5c2942306a7b1bb4a69
```

**Pull request merged**